### DISCUSSION TITLE
modernizr.com status

### DISCUSSION THREAD

**Discussion by rejas** [Modernizr/Modernizr#2635] [General]
Hi,
with reports of modernizrs homepage being down or outdated coming into the system every now and then its time to tackle this issue for good. I hope @patrickkettner reads this and can come here too to discuss the future of the website. Also @Markel and  @roblarsen who still help with the project too.

Should we maintain it and how? 
Who owns the domain and how are the costs?
Are all features still necessary (I am looking at you, bower)

IMHO the website is stil useful and should be maintained. Not with new features but at least with new releases and release notes.

Happy to hear from you all

**Comment by Markel** [comment]
Oh boy, I feel that this is going to be a long message.

I totally agree that the website should not be taken down or make it a redirect of some kind to the repository, as I actually think that providing the library with a nice UX/UI is useful specially to beginners (I actually I'm here ~4 years later helping because the website got stuck with me, even though I never got to implement it :sweat_smile:).

But yeah, we should rely on the website less (therefore reducing the costs). To my understanding [there are actually two servers: the one which builds modernizr and the one that serves the HTML](https://github.com/Modernizr/Modernizr/pull/2481#issuecomment-618745541) (correct me on that one if I'm wrong @patrickkettner), so I've been thinking that we should try users to build modernizr custom builds themselves ("getting rid" of the build server) and document it (I think that it is possible right now but who knows, as I said I never archived nor attempted since to use Modernizr :rofl:), so instead of giving them the .js file we give them a command with simple instructions so they only have to copy paste (the command would be created on the frontend), similar to the current command line config option but simpler or with better documentation. The problem I see to this approach is that new detects and description changes would still need a website update to appear<sup><b>1</b></sup>, but honestly, that may be the lesser of the problems. Also, I guess they would need to have npm installed (that one is worse, but I guess nice documentation could help with it).

About Bower, I mean, if literally their main page says that it is not recommended to use it... I feel like we are wasting a little bit of time if we continue to support it + generating costs to the server, however if I have learnt something today is that [there's still people using it](https://github.com/Modernizr/Modernizr/issues/2634#issue-789159601) so we might have to further consider this. However, as we're approaching the v4 release #2594 (I hope I don't jinx it) we could implement this "degradation of service" as something natural from a mayor release. The v4 release can also allow us to implement things like an easier command line option (why am I giving myself work?) or other things that could make things easier to maintain.

But everything said here is subject to change depending on the answer to the questions that @rejas proposes, we obviously need the support from someone who at least can control the DNS and maybe the server (depending on our approach), but yeah, seeing how things have developed in the last 2 years if we are going to do something about the website my objectives would be these 2: 

- Reducing the maintenance needed
- Reducing the costs (because spending less money is always better)

**1**: If we reduced the website to just a front-end face (or a script that generates an static website) we could build it directly every release (therefore solving the update problems) with Github Actions (don't kill me @rejas) or other tools and host it in one of the dozen platforms that allows for it. Moreover, it would allow to integrate other things much easier to the web.

But yeah, this is the "summary" of the conclusion I've reached since #2628 was opened, as always any ideas, suggestions and comments are more than welcome :)

**Comment by roblarsen** [comment]
If it's just static, hosting on gh-pages is the easiest option. That's what we do with html5boilerplate.com

**Comment by patrickkettner** [comment]
its not currently static, because a fair amount of people use the builder.
I built a version on netlify functions that would keep it working while
also being a static version, but it was blocked on the v4 issue. If that
gets figured out, then v4 can release and the site can be automatically
updated on push

On Fri, Jan 29, 2021 at 11:40 AM Rob Larsen <notifications@github.com>
wrote:

> If it's just static, hosting on gh-pages is the easiest option. That's
> what we do with html5boilerplate.com
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Modernizr/Modernizr/discussions/2635#discussioncomment-321262>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADRUBXP5O6XPN5JVSLYVWLS4LQGLANCNFSM4WJA7Y3A>
> .
>


-- 
patrick

**Reply by rejas** [reply]
What is the blocking issue for the new page exactly?

**Comment by patrickkettner** [comment]
Nothing is blocking the new page. The build function of v4 needs to be
redone, it has recursive loop that means it builds absurdly slow/ OOMs.
Everything should be documented in the PR

On Sat, Jan 30, 2021 at 11:52 AM Veeck <notifications@github.com> wrote:

> What is the blocking issue for the new page exactly?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Modernizr/Modernizr/discussions/2635#discussioncomment-323578>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADRUBTBXOTWM2EZTAV6YGTS4Q2M7ANCNFSM4WJA7Y3A>
> .
>
-- 
patrick

**Comment by tinatenatean** [comment]
Hi:  I understand that the Modernizr.com build is outdated.  How do I go about adding only the custom features I want into my Visual Studio project?  Thanks for any clarification.

**Reply by rejas** [reply]
You can always build a modernizr file for yourself by using the cli, see https://github.com/modernizr/modernizr#from-the-command-line

**Reply by rejas** [reply]
Be sure to check out the v3 branch though and not the master branch (which is still work in progress)
